### PR TITLE
soapy: consistency in input validation in new functions

### PR DESCRIPTION
**Pull request by ncorgan**
* Refactored out input validation functions for later use

Signed-off-by: Nicholas Corgan <n.corgan@gmail.com>

### CODE REVIEW

File: gr-soapy/lib/block_impl.cc [hunk 1]
```diff
@@ -28,6 +28,57 @@ static bool arg_info_has_key(const arginfo_list_t& info_list, const std::string&
     });
 }
 
+template <typename T>
+static inline bool vector_contains(const std::vector<T>& vec, const std::string& elem)
+{
+    return (std::find(vec.begin(), vec.end(), elem) == vec.end());
```
Comment (willcode) [inline gr-soapy/lib/block_impl.cc:34]: `!=`?

Comment (ncorgan) [reply gr-soapy/lib/block_impl.cc]: Fixed.

**Comment by willcode** [comment]
Not totally related to this PR: what can we do to make some of these messages more useful to the user, for example when a discrete set of rates is available (frequency, bandwidth), something that a lot of users are going to do:
```
In [4]: b.set_sample_rate(0,5)                                                                                          
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-b7072cf44603> in <module>
----> 1 b.set_sample_rate(0,5)

ValueError: source: Unsupported sample rate (5.000000).  Rate must be in the range [1000000.000000, 1000000.000000], [2000000.000000, 2000000.000000], [3000000.000000, 3000000.000000], [4000000.000000, 4000000.000000], [5000000.000000, 5000000.000000], [6000000.000000, 6000000.000000], [7000000.000000, 7000000.000000], [8000000.000000, 8000000.000000], [9000000.000000, 9000000.000000], [10000000.000000, 10000000.000000], [11000000.000000, 11000000.000000], [12000000.000000, 12000000.000000], [13000000.000000, 13000000.000000], [14000000.000000, 14000000.000000], [15000000.000000, 15000000.000000], [16000000.000000, 16000000.000000], [17000000.000000, 17000000.000000], [18000000.000000, 18000000.000000], [19000000.000000, 19000000.000000], [20000000.000000, 20000000.000000]
```

Or when a function is not supported:
```
In [5]: b.set_master_clock_rate(5)                                                                                      
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-5-10308577ca2e> in <module>
----> 1 b.set_master_clock_rate(5)

IndexError: basic_string::erase: __pos (which is 18446744073709551614) > this->size() (which is 0)
```
```
In [21]: b.set_clock_source('EXT')                                                                                      
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-21-7f272b15f1be> in <module>
----> 1 b.set_clock_source('EXT')

IndexError: basic_string::erase: __pos (which is 18446744073709551615) > this->size() (which is 1)
```

**Comment by ncorgan** [comment]
For that first one, we'd have to specifically detect that case and put out a special message. This is doable but adds some processing time when this happens. This might be fine, as we're already in an error state anyway, so it doesn't block the critical path.

The second two are string processing bugs; I'll push up a fix shortly.

**Comment by willcode** [comment]
Since this is just error code, how about having range_to_string return minimum() if minimum()==maximum()? That would make the result of ranges_to_string() readable.

**Comment by willcode** [comment]
OK, I think this one is good.

If you get a chance, not critical, in `soapy_types_python.cc`, extra paren in `range_t.__str__()`:
```
In [12]: str(b.get_frequency_range(0)[0])                                                                               
Out[12]: '(minimum: )0.000000, maximum: 7250000000.000000, step: 0.000000)'
```

**Pull request merged**